develop/smart-contracts/precompiles/xcm-precompile.md

```diff
@@ -0,0 +1,120 @@
+
+The function returns a `Weight` struct containing `refTime` and `proofSize` values, which indicate the estimated computational cost of executing this message. If successful, after calling the `callWeighMessage` function, you should see the `refTime` and `proofSize` of the message:
+
+![](/images/develop/smart-contracts/precompiles/xcm-precompile/xcm-precompile-05.webp)
+
+!!!note
+    For example, to interact with Polkadot Hub TestNet, you can check this [gist](https://gist.github.com/franciscoaguirre/a6dea0c55e81faba65bedf700033a1a2){target=\_blank}, which provides examples of how to craft XCM messages for different purposes.
+
+### Execute a Message
+
+The `execute` function runs an XCM message locally using the caller's origin. This is useful for executing XCM instructions that don't require cross-chain communication.
+
+To execute a message:
+
+1. Call `callWeighMessage` with your XCM message to get the required weight
+2. Use the returned weight values when calling `callXcmExecute`
+3. Pass the same XCM message bytes and the weight obtained from the previous step. For example, using the same message from the weighing example, you would call `callXcmExecute` with:
+
+    - `message`: The encoded XCM message bytes.
+    - `weight`: The `Weight` struct returned from `callWeighMessage`.
+
+    You can examine the complete extrinsic structure for this operation [here](https://dev.papi.how/extrinsics#networkId=localhost&endpoint=wss%3A%2F%2Ftestnet-passet-hub.polkadot.io&data=0x1f03050c000401000003008c86471301000003008c8647000d010101000000010100368e8759910dab756d344995f1d3c79374ca8f70066d3a709e48029f6bf0ee7e0750c61e2901daad0600){target=\_blank}.
+
+5. Click on the **Transact** button to execute the xcm message:
+  
+    ![](/images/develop/smart-contracts/precompiles/xcm-precompile/xcm-precompile-06.webp)
+
+If successful, you will see the following output in the Remix terminal:
+
+![](/images/develop/smart-contracts/precompiles/xcm-precompile/xcm-precompile-07.webp)
+
+Additionally, you can verify that the execution of this specific message was successful by checking that the beneficiary account associated with the xcm message has received the funds accordingly.
+
+### Send a Message
+
+The `send` function transmits an XCM message to a destination chain. This is the core functionality for cross-chain communication.
+
+To send a message:
+
+1. Prepare your destination location encoded in XCM format
+2. Prepare your XCM message (similar to the execute example)
+3. Call `callXcmSend` with both parameters
+
+The destination parameter must be encoded according to XCM's location format, specifying the target parachain or consensus system. The message parameter contains the XCM instructions to be executed on the destination chain.
+
+Unlike `execute`, the `send` function doesn't require a weight parameter since the destination chain will handle execution costs according to its own fee structure.
```

Comment:
```suggestion
Unlike `execute`, the `send` function doesn't require a weight parameter since the destination chain will handle execution costs according to its fee structure.
```